batch/batch/api.py

```diff
@@ -3,62 +3,109 @@
 import random
 import requests
 
+class API():
+    def __init__(self, timeout=None):
+        self.timeout = timeout
+
+    def create_job(url, spec, attributes, batch_id, callback):
+        d = {'spec': spec}
+        if attributes:
+            d['attributes'] = attributes
+        if batch_id:
+            d['batch_id'] = batch_id
+        if callback:
+            d['callback'] = callback
+
+        r = requests.post(url + '/jobs/create', json = d, timeout = self.timeout)
+        r.raise_for_status()
+        return r.json()
+
+    def list_jobs(url):
+        r = requests.get(url + '/jobs', timeout = self.timeout)
+        r.raise_for_status()
+        return r.json()
+
+    def get_job(url, job_id):
+        r = requests.get(url + '/jobs/{}'.format(job_id), timeout = self.timeout)
+        r.raise_for_status()
+        return r.json()
+
+    def get_job_log(url, job_id):
+        r = requests.get(url + '/jobs/{}/log'.format(job_id), timeout = self.timeout)
+        r.raise_for_status()
+        return r.text
+
+    def delete_job(url, job_id):
+        r = requests.delete(url + '/jobs/{}/delete'.format(job_id), timeout = self.timeout)
+        r.raise_for_status()
+        return r.json()
+
+    def cancel_job(url, job_id):
+        r = requests.post(url + '/jobs/{}/cancel'.format(job_id), timeout = self.timeout)
+        r.raise_for_status()
+        return r.json()
+
+    def create_batch(url, attributes):
+        d = {}
+        if attributes:
+            d['attributes'] = attributes
+        r = requests.post(url + '/batches/create', json = d, timeout = self.timeout)
```

Comment:
PEP style prefers no spaces around the `=` in kwargs, but I don't really care.

So far I've tried to use this style for everything that isn't Hail annotate_... calls.